code/modules/mob/living/carbon/xenomorph/xeno_helpers.dm

```diff
@@ -13,8 +13,18 @@
 	SHOULD_CALL_PARENT(FALSE)
 	return FALSE
 
+/mob/living/carbon/xenomorph/get_crit_threshold()
+	. = ..()
+	if(!endure)
+		return
+	var/datum/action/ability/xeno_action/endure/endure_ability = actions_by_path[/datum/action/ability/xeno_action/endure]
+	return endure_ability.endure_threshold
+
 /mob/living/carbon/xenomorph/get_death_threshold()
-	return xeno_caste.crit_health
+	if(!endure)
+		return xeno_caste.crit_health
+	var/datum/action/ability/xeno_action/endure/endure_ability = actions_by_path[/datum/action/ability/xeno_action/endure]
+	return endure_ability.endure_threshold
```

Comment:
maybe out of scope, but all these funny xeno mob action checks should presumably be handled via signals

Reply:
shrug prolly